homeassistant/components/sensor/discogs.py

```diff
@@ -40,32 +43,75 @@
     """Set up the Discogs sensor."""
     import discogs_client
 
-    name = config.get(CONF_NAME)
-    token = config.get(CONF_TOKEN)
+    token = config[CONF_TOKEN]
 
     try:
-        discogs = discogs_client.Client(SERVER_SOFTWARE, user_token=token)
-        identity = discogs.identity()
+        discogs_client = discogs_client.Client(
```

Comment:
I'm now passing along the client and acting on the identity in the update methods instead of getting the identity here and passing that along